cli/src/main/java/org/jboss/as/cli/CommandLineMain.java

```diff
@@ -151,4 +158,29 @@ private static Properties createLogManagerConfig(final String level) {
         properties.setProperty("formatter.PATTERN.properties", "pattern");
         return properties;
     }
+
+    /**
+     * Creates a file marker under $JBOSS_HOME/bin directory. This is used by the shutdown operation to understand from where we have
+     * launched CLI instance when we are performing an update of a remote installation.
+     */
+    private static void createClientMarker() {
+        try {
+            final String jbossHome = WildFlySecurityManager.getEnvPropertyPrivileged("JBOSS_HOME", null);
```

Comment:
I am not sure we need the SecurityManager wrappers here, in this case as this is the entry point there should not be other protection domains on the call stack.

Reply:
I am suggesting neither, just access the property directly.

Reply:
The security manager being possibly enabled is never a reason to add a doPrivileged.

When you add a doPrivileged call what you are effectively doing is dropping the protection domains of any callers currently on the call stack and setting up the permission check to be in the context of the protection domain of the class that is calling the method with the permissions check.

As this is a main method / entry point - what callers are being dropped from the call stack and what measures have been put in place to verify they can safely call this method.

The way these properties are being used the safety is probably fine but what callers are being dropped from the call stack?

Reply:
> I just wanted to avoid the need to require specific permission to load an environment variable that is indeed configured by the launch script. We know that this is safe to do because it is the launch script the one that invokes this Main class.

This is not the purpose of a doPrivileged call, it is not possible to prevent the permission check.  The ProtectionDomain for the CommandLineMain class will still need the relevant permissions to be granted.

When a permission check is performed it loops through the PermissionDomains of all classes on the call stack, and each and every one of those must be granted that permission or the check will fail.  What doPrivileged does is it says ignore the classes that called me but it does not get away from the "me" class being required the permission.

If this class is always intended to be called as the main entry point to the program it does not need any doPrivileged calls as it will always be the starting point of the call stack so there will be no protection domains to drop.

On the other hand if this is expected that other classes from other protection domains will call this class then this class should only be using doPrivileged if those are trusted called / it has been designed so the doPrivileged calls can not be abused.

TBH this pattern that creeps in where if there is going to be a permission check a doPrivileged is added is one of the reasons that adds to the justification of the removal of the security manager in Java.

Reply:
I'm starting to understand the tricky details of the security manager when, for good reasons, it was going to be removed :)...never is late, even if it going to be an obsolete knowledge